python/ray/ml/trainer.py

```diff
@@ -66,7 +85,6 @@ def fit(self) -> Result:
         """
         raise NotImplementedError
 
-    @abc.abstractmethod
```

Comment:
should either be abstract or implemented.

It's great that we unified Trainer and FunctionTrainer, but how do we explain/justify the existence of both "training_function" and "as_trainable" now? Do we recommend "training_function" as the standard way of doing things? And if you know what a Trainable is, you can also use "as_trainable"?

Reply:
@maxpumperla yes that’s right users are expected to only override the abstract methods (training_function).

as_trainable will contain an implementation, it’s just out of scope for this PR.